exp/api/v1beta2/rosamachinepool_types.go

```diff
@@ -105,6 +105,11 @@ type RosaMachinePoolSpec struct {
 	//
 	// +optional
 	NodeDrainGracePeriod *metav1.Duration `json:"nodeDrainGracePeriod,omitempty"`
+
+	// UpgradeManagementConfig specifies upgrade management configuration which set MaxUnavailable & MaxSurge number of nodes during upgrade.
+	//
+	// +optional
+	UpgradeManagementConfig *RosaMachinePoolUpgradeConfig `json:"upgradeManagementConfig,omitempty"`
```

Comment:
```suggestion
	RollingUpdate *RosaRollingUpdateConfig  `json:"rollingUpdate,omitempty"`
```

Reply:
I'm not against  the naming but in OCM lib it is NodePoolManagementUpgrade, in hyperShift it is NodePoolManagement. Also it may confuse with the rollingUpgrade for other CRs. I'm more to keep the same name.

Reply:
OCM naming is irrelevant. and its also `rollingUpdate` in Hypershift (https://github.com/openshift/hypershift/blob/afd17d6a127d4bcffe31a063af988973b495b0e1/api/hypershift/v1beta1/nodepool_types.go#L277). its just wrapped within other fields.

> Also it may confuse with the rollingUpgrade for other CRs

This is what its called in CAPI, so CAPI users will know immediately what the field does.

Reply:
right, so its wrapped in another field in hypershift as the upgrade Type is included [here](https://github.com/openshift/hypershift/blob/afd17d6a127d4bcffe31a063af988973b495b0e1/api/hypershift/v1beta1/nodepool_types.go#L356) (ocm had the management name as all fields together) , However I didn't expose that to CAPA as its only support Replace type for now.

Reply:
exactly, we are not exposing any of the other fields. We are only exposing `RollingUpdate` fields, so we need to name it accordingly:

again, the field is named `RollingUpdate` in hypershift (no that it matters): https://github.com/openshift/hypershift/blob/afd17d6a127d4bcffe31a063af988973b495b0e1/api/hypershift/v1beta1/nodepool_types.go#L277-L282

but most importantly is named in CAPI `RollingUpdate` https://github.com/kubernetes-sigs/cluster-api/blob/6941e8c66f3de69adfb1f19893afc0f3cf4571e6/api/v1beta1/machinedeployment_types.go#L164

Once we have support for more fields, update strategies, etc. Then we can think about wrapping this in another upper management field.